e2e/cypress/tests/integration/messaging/forward_message_from_dm_spec.js

```diff
@@ -0,0 +1,220 @@
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @enterprise @messaging
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Forward Message', () => {
+    let user1;
+    let user2;
+    let testTeam;
+    let dmChannel;
+    let testPost;
+    let replyPost;
+
+    const message = 'Forward this message';
+    const replyMessage = 'Forward this reply';
+    const commentMessage = 'Comment for the forwarded message';
+
+    /**
+     * Verify that the post has been forwarded
+     *
+     * @param {string?} comment
+     * @param {boolean?} showMore
+     * @param {Post} post
+     */
+    const verifyForwardedMessage = ({post, comment, showMore}) => {
+        const permaLink = `${Cypress.config('baseUrl')}/${testTeam.name}/pl/${post.id}`;
+
+        // * Assert post has been forwarded
+        cy.getLastPostId().then((id) => {
+            // * Assert last post is visible
+            cy.get(`#${id}_message`).should('be.visible').within(() => {
+                if (comment) {
+                    // * Assert the text in the post body is the permalink only
+                    cy.get(`#postMessageText_${id}`).should('be.visible').should('contain.text', permaLink).should('contain.text', comment);
+
+                    if (showMore) {
+                        // * Assert show more button is rendered and works as expected
+                        cy.get('#showMoreButton').should('be.visible').should('contain.text', 'Show more').click().should('contain.text', 'Show less').click();
+                    }
+                }
+
+                // * Assert there is only one preview element rendered
+                cy.get('.attachment.attachment--permalink').should('have.length', 1);
+
+                // * Assert the text in the preview matches the original post message
+                cy.get(`#postMessageText_${post.id}`).should('be.visible').should('contain.text', post.message);
+            });
+
+            // # Cleanup
+            cy.apiDeletePost(id);
+        });
+    };
+
+    /**
+     * Forward Post with optional comment.
+     * Has the possibility to also test for the post-error on long comments
+     *
+     * @param {object?} options
+     * @param {string?} options.comment
+     * @param {boolean?} options.cancel
+     */
+    const forwardPostFromDM = ({comment = '', cancel = false} = {}) => {
+        // * Assert visibility of the forward post modal
+        cy.get('#forward-post-modal').should('be.visible').within(() => {
+            // * Assert channel select is not existent
+            cy.get('.forward-post__select').should('not.exist');
+
+            // * Assert if button is enabled
+            cy.get('.GenericModal__button.confirm').should('not.be.disabled');
+
+            // * Assert Notificatio is shown
```

Comment:
Typo. Also in other places.
```suggestion
            // * Assert Notification is shown
```